### ISSUE TITLE
Crashing when editing user preferences 

### ISSUE THREAD

**Issue by olivif** [Microsoft/vscode#1570]
I'm seeing a lot of crashes and hangs in 0.10.5 when editing user preferences (and a few other hangs here and there when editing regular files). 

I can repro this quite easily so if there's any help you need getting some logs/data locally just let me know. Repro steps usually are:
1. Open vscs
2. Open File\Preferences\User settings
3. Edit user settings file - copy from template, paste into user settings, keep copy and pasting
4. vscs hangs, eventually get a not responding from Windows

I've seen a few other hangs here and there when editing regular files but I can't really tell if they're related. 
 
Here are my version stats
![image](https://cloud.githubusercontent.com/assets/7736961/11951653/f573d616-a892-11e5-9217-bfddeeae5b4a.png)
![image](https://cloud.githubusercontent.com/assets/7736961/11951716/7f073a12-a893-11e5-8043-fe17b8123425.png)

**Comment by olivif** [comment]
Interestingly enough, after disabling all extensions, the issue cannot be reproed anymore. I suspect the issue was related to the JSCS linter, as the text I was editing in the user preferences file was also JSCS linter related.

**Comment by bpasero** [comment]
The PH and any extension inside should not cause the renderer to crash.

**Comment by olivif** [comment]
@bpasero thanks for assigning! :smile: good point, the extensions shouldn't crash the renderer. I tried to repro again and watch the memory in task manager but can't seem to repro anymore.

**Comment by alexandrudima** [comment]
cc @chrisdias @dbaeumer 

@olivif Do you have more details or maybe you remember something else you did, I tried unsuccessfully to repro by doing the following:

* installed jscs extension
* `npm install jscs` in the folder I have opened
* copied the entire default user settings
* created a new `test.js` file
* pasted many times, 
* reached 40k lines
* I see no crash

Is there more to it?

![image](https://cloud.githubusercontent.com/assets/5047891/12151825/53a3e710-b4b1-11e5-9eb2-639461bc00e8.png)

**Comment by olivif** [comment]
@alexandrudima thanks for having a look. those are the steps I did as well, the main difference to your example is that I was editing jscs preferences, specifically `"jscs.configuration": ""`. The other difference is that I was doing: copy, paste, save, remove some lines (ctrl X), save, repeat.  

maybe it was a combination of extensions I had installed with the preferences I was editing. regardless, after removing all extensions I couldn't repro anymore. I have also upgraded to 0.10.6, reinstalled jscs and tried to repro again with no luck. 

the strange thing was that it was super consistent at that point, I could just follow the steps and repro every time. after reinstalling extensions and upgrading, I couldn't repo even once.
